scripts/xtensa-build-zephyr.py

```diff
@@ -0,0 +1,382 @@
+						help="Initial Zephyr git ref for the -c option. Can be a branch, tag,"
+						"full SHA1 in a fork")
+	parser.add_argument("-u", "--url", required=False,
+						default="https://github.com/zephyrproject-rtos/zephyr/",
+						help="URL to clone Zephyr from")
+	mode_group = parser.add_mutually_exclusive_group()
+	mode_group.add_argument("-p", "--west_path", required=False, type=pathlib.Path,
+						help="Existing Zephyr project directory. Incompatible with -c."
+						"If zephyr-project/modules/audio/sof is missing then a"
+						"symbolic link pointing to ${SOF_TOP} will automatically be"
+						"created and west will recognize it as its new sof module."
+						"If zephyr-project/modules/audio/sof already exists and is a"
+						"different copy than where this script is run from, then the"
+						"behavior is undefined."
+						"This -p option is always required_if the real (not symbolic)"
+						"sof/ and zephyr-project/ directories are not nested in one"
+						"another.")
+	mode_group.add_argument("-c", "--clone", required=False, action="store_true",
+						help="Using west, downloads inside this sof clone a new Zephyr"
+						"project with the required git repos. Creates a"
+						"sof/zephyrproject/modules/audio/sof symbolic link pointing"
+						"back at this sof clone."
+						"Incompatible with -p. To stop after downloading Zephyr, do not"
+						"pass any platform or cmake argument.")
+	parser.add_argument("-v", "--verbose-build", required=False, action="store_true",
+						help="Verbose build system log")
+	parser.add_argument("-vv", "--more-verbose", required=False, action="store_true",
+						help="Implicitly adds -v and additionally prints commands executed by"
+						" this script")
+	parser.add_argument("-C", "--cmake-args", required=False,
+						help="Cmake arguments passed to cmake configure step without\n"
+						"'-D' prefix. Eg. CMAKE_GENERATOR='Ninja'. Preserves internal quotes.")
+
+	args = parser.parse_args()
+
+	if args.all:
+		args.platforms = platform_names
+
+	# print help message if no arguments provided
+	if len(sys.argv)==1:
+			parser.print_help()
+			sys.exit(0)
+
+	if args.z and not args.clone:
+		raise RuntimeError(f"Argument -z without -c makes no sense")
+	if args.clone and not args.z:
+		args.z = "main"		# set default name for -z if -c specified
+
+	if args.west_path:
+		WEST_TOP=pathlib.Path(args.west_path)
+
+	if args.more_verbose:
+		args.verbose_build = True
+
+	if args.verbose_build:
+		if not args.cmake_args:
+			args.cmake_args = "CMAKE_VERBOSE_MAKEFILE=ON"
+		else:
+			args.cmake_args = "CMAKE_VERBOSE_MAKEFILE=ON " + args.cmake_args
+
+	# split arguments by whitespaces and append -D prefix to match expected CMake format
+	if args.cmake_args:
+		arg_list = shlex.split(args.cmake_args)
+		arg_list = ["-D" + arg for arg in arg_list]
+		args.cmake_args = " ".join(arg_list)
+
+def execute_command(command_args, stdin=None, input=None, stdout=None, stderr=None,
```

Comment:
```suggestion
def execute_command(*cmd, **kwargs)
```

Reply:
No thank you, this assumes many things that may not be true.
As I understand this approach, you simply pass a list of all arguments in *cmd list.
Meanwhile, when you use subprocess.run you can pass selective arguments by name without maintaining the order or arguments.
Even subprocess run signature:
`def run(*popenargs,
        input=None, capture_output=False, timeout=None, check=False, **kwargs):`
does not throw all args into single array. And then *popenargs is passed to Popen ctor to resolve what is where.
I need to log cwd of the command, why would assume its position in the *cmd arguments list?

Reply:
`subprocess.run()` and `.popen()` have a single positional argument (which can be either a string a or a list). Maybe this would work with `def execute_command(cmd, **kwargs)`, I don't know.